store/tikv/2pc.go

```diff
@@ -437,13 +437,18 @@ func (c *twoPhaseCommitter) initKeysAndMutations() error {
 			value = it.Value()
 			if len(value) > 0 {
 				if tablecodec.IsUntouchedIndexKValue(key, value) {
-					continue
-				}
-				op = pb.Op_Put
-				if flags.HasPresumeKeyNotExists() {
-					op = pb.Op_Insert
+					if !flags.HasLocked() {
+						continue
+					}
+					op = pb.Op_Lock
```

Comment:
So will it hurt the commit performance? like,
```sql
begin;
update t set i = i;
commit;
```

Reply:
`update t set i = i`'s mutations do not have value, and the pessimistic locks created by it are also resolved before this PR.
https://github.com/pingcap/tidb/blob/183f2f3f97e4abf5beadd7f759d2970d48b3af8c/store/tikv/2pc.go#L430-L435

Reply:
We need to add some cases to validate the primary key processing in `2pc_test`.It's dangerous if we could not hanlde the newly added keys properly.